svelte/src/lib/components/support/CrateReportForm.svelte

```diff
@@ -0,0 +1,308 @@
+
+  // svelte-ignore state_referenced_locally
+  let crate = $state(initialCrate);
+  let selectedReasons = $state<string[]>([]);
+  let detail = $state('');
+
+  let crateInvalid = $state(false);
+  let reasonsInvalid = $state(false);
+  let detailInvalid = $state(false);
+
+  let isMaliciousCodeReport = $derived(selectedReasons.includes('malicious-code'));
+  let isVulnerabilityReport = $derived(selectedReasons.includes('vulnerability'));
+
+  function validate(): boolean {
+    crateInvalid = !crate || !crate.trim();
+    reasonsInvalid = selectedReasons.length === 0;
+    detailInvalid = selectedReasons.includes('other') && !detail?.trim();
+    return !crateInvalid && !reasonsInvalid && !detailInvalid;
+  }
+
+  function isReasonSelected(reason: string): boolean {
+    return selectedReasons.includes(reason);
+  }
+
+  function toggleReason(reason: string): void {
+    selectedReasons = selectedReasons.includes(reason)
+      ? selectedReasons.filter(it => it !== reason)
+      : [...selectedReasons, reason];
+    reasonsInvalid = false;
+  }
+
+  function composeMail(): string {
+    let reasons = REASONS.map(({ reason, description }) => {
+      let selected = isReasonSelected(reason);
+      return `${selected ? '- [x]' : '- [ ]'} ${description}`;
+    }).join('\n');
+
+    let body = `I'm reporting the https://crates.io/crates/${crate} crate because:
+
+${reasons}
+
+Additional details:
+
+${detail}
+`;
+    let subject = `The "${crate}" crate`;
+    if (isMaliciousCodeReport) {
+      subject = `[SECURITY] ${subject}`;
+    }
+
+    let addresses = 'help@crates.io';
+    if (isMaliciousCodeReport) {
+      addresses += ',security@rust-lang.org';
+    }
+
+    return `mailto:${addresses}?subject=${encodeURIComponent(subject)}&body=${encodeURIComponent(body)}`;
+  }
+
+  function handleSubmit(event: SubmitEvent): void {
+    event.preventDefault();
+
+    if (!validate()) {
+      return;
+    }
+
+    let mailto = composeMail();
+    window.open(mailto, '_self');
+  }
+</script>
+
+<form {...restProps} class="report-form" data-testid="crate-report-form" onsubmit={handleSubmit}>
+  <h2>Report A Crate</h2>
+
+  <fieldset class="form-group" data-test-id="fieldset-crate">
+    <label for="{id}-crate" class="form-group-name">Crate</label>
+    <!-- svelte-ignore a11y_autofocus -->
+    <input
+      id="{id}-crate"
+      type="text"
+      bind:value={crate}
+      autocomplete="off"
+      aria-required="true"
+      aria-invalid={crateInvalid}
+      class="crate-input base-input"
+      data-test-id="crate-input"
+      autofocus
+      oninput={() => (crateInvalid = false)}
+    />
+    {#if crateInvalid}
+      <div class="form-group-error" data-test-id="crate-invalid">Please specify a crate.</div>
+    {/if}
+  </fieldset>
+
+  <fieldset class="form-group" data-test-id="fieldset-reasons">
+    <div class="form-group-name">Reasons</div>
+    <ul role="list" class="reasons-list scopes-list" class:invalid={reasonsInvalid}>
```

Comment:
```suggestion
    <ul role="list" class={["reasons-list scopes-list", { invalid: reasonsInvalid }]}>
```

I'm suggesting this primarily because of the following note in the official documentation.
> Unless you’re using an older version of Svelte, consider avoiding class:, since the attribute is more powerful and composable. [^1]

[^1]: https://svelte.dev/docs/svelte/class#The-class:-directive

Reply:
> also, Claude prefers it this way... 😆😅

I assume this is because the attribute-based approach is a new feature that needs a bit more time to mature.

---

Your point is fair, though. I'm not strongly opposed to the old way, and we can always retrofit it later if needed!